include/onnxruntime/core/common/common.h

```diff
@@ -42,6 +42,14 @@
 
 namespace onnxruntime {
 
+#ifdef USE_MIMALLOC
+template <typename T>
+using FastAllocVector = std::vector<T,mi_stl_allocator<T>>;
+#else
+template <typename T>
+using FastAllocVector = std::vector<T>;
+#endif
+
 using TimePoint = std::chrono::high_resolution_clock::time_point;
```

Comment:
This is a very global place. If FastAllocVector is just going to be used in CPU kernels, perhaps a file under onnxruntime/core/providers/cpu would be better?